spec/views/_webchat_message.html.erb_spec.rb

```diff
@@ -0,0 +1,31 @@
+require 'spec_helper'
+
+RSpec.describe 'webchat_message/_footer', type: :view do
+  shared_examples_for 'a webchat link' do |partial_view|
+    subject(:rendered_page) do
+      render partial_view
+
+      rendered
+    end
+
+    context 'when ENV WEBCHAT_URL is set' do
+      before do
+        ENV['WEBCHAT_URL'] = 'http://webchat_url_test'
```

Comment:
This changes this env var for all subsequent specs as well as this one. Probably not an issue for this particlar spec but this pattern can introduce flaky specs since the randomised ordering of specs will affect whether this breaks other subsequent specs?

A different approach is to use mocking?

```
before do
  allow(ENV).to receive(:[]).and_call_original
  allow(ENV).to receive(:[]).with('WEBCHAT_URL').and_return('http://webchat_test_url')
end
```

Reply:
Or even easier is stubbing the ruby code used to access the env var

```ruby
before { allow(TradeTariffFrontend).to receive(:webchat_url).and_return('http://webchat_test_url') }
```